src/main/docker/Dockerfile.jvm.staged

```
@@ -29,42 +29,25 @@ COPY src src
+RUN grep version /build/target/maven-archiver/pom.properties | cut -d '=' -f2 >.env-version
 RUN grep artifactId /build/target/maven-archiver/pom.properties | cut -d '=' -f2 >.env-id
 
-# mv the uber jar in case the quarkus property quarkus.package.type=uber-jar is set
-RUN mv /build/target/$(cat .env-id)-$(cat .env-version)*.jar /build/target/quarkus-app-tmp/
-
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.7-1085 
-
-ARG JAVA_PACKAGE=java-17-openjdk-devel
-ARG RUN_JAVA_VERSION=1.3.8
-ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
-# Install java and the run-java script
-# Also set up permissions for user `1001`
-RUN microdnf install curl ca-certificates ${JAVA_PACKAGE} \
-    && microdnf update \
-    && microdnf clean all \
-    && mkdir /deployments \
-    && chown 1001 /deployments \
-    && chmod "g+rwX" /deployments \
-    && chown 1001:root /deployments \
-    && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /deployments/run-java.sh \
-    && chown 1001 /deployments/run-java.sh \
-    && chmod 555 /deployments/run-java.sh \
-    && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
+# if this is an uber jar create a structure that looks the same as fast-jar with empty directories
+# this allows for the same dockerfile to be used with both
+RUN if [ ! -d /build/target/quarkus-app ] ; then mkdir -p /build/target/quarkus-app/lib; \
+     mkdir -p /build/target/quarkus-app/app; \
+     mkdir -p /build/target/quarkus-app/quarkus; \
+     mv /build/target/$(cat .env-id)-$(cat .env-version)*.jar /build/target/quarkus-app/ ; \
+     fi
 
+FROM registry.access.redhat.com/ubi8/openjdk-17-runtime:1.15
 # Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
 ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Dquarkus.http.port=8081 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 # We make four distinct layers so if there are application changes the library layers can be re-used
-COPY --from=0 --chown=1001 /build/target/quarkus-app-tmp/*.jar /deployments/export-run-artifact.jar
-COPY --from=0 --chown=1001 /build/target/*quarkus-app/*lib/ /deployments/lib/
-# Overwrite the uber jar if package type is normal
-COPY --from=0 --chown=1001 /build/target/*quarkus-app/*.jar /deployments/export-run-artifact.jar 
-COPY --from=0 --chown=1001 /build/target/*quarkus-app/*app/ /deployments/app/
-COPY --from=0 --chown=1001 /build/target/*quarkus-app/*quarkus/ /deployments/quarkus/
-
+COPY --from=0 --chown=1001 /build/target/*quarkus-app/lib/ /deployments/lib/
+COPY --from=0 --chown=1001 /build/target/*quarkus-app/*.jar /deployments/export-run-artifact.jar
+COPY --from=0 --chown=1001 /build/target/*quarkus-app/app/ /deployments/app/
+COPY --from=0 --chown=1001 /build/target/*quarkus-app/quarkus/ /deployments/quarkus/
```

```suggestion
COPY --from=0 --chown=1001 /build/target/quarkus-app/lib/ /deployments/lib/
COPY --from=0 --chown=1001 /build/target/quarkus-app/*.jar /deployments/export-run-artifact.jar
COPY --from=0 --chown=1001 /build/target/quarkus-app/app/ /deployments/app/
COPY --from=0 --chown=1001 /build/target/quarkus-app/quarkus/ /deployments/quarkus/
```

since we are making this dir, we can remove the `*` I think